From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nick Desaulniers <ndesaulniers@google.com>
Date: Wed, 17 Mar 2021 13:54:58 -0700
Subject: ANDROID: re-enable BIG_ENDIAN in allmodconfig testing

Revert "ANDROID: Fix arm64 allmodconfig build"
This reverts commit 099f0261d93727c97aa0dea2ed22bedf991fe574.

Proper KCONFIG checks were add in

commit e9c6deee00e9 ("arm64: Make CPU_BIG_ENDIAN depend on ld.bfd or
ld.lld 13.0.0+")

in v5.12-rc1 for ARCH=arm64, and

commit 28187dc8ebd9 ("ARM: 9025/1: Kconfig: CPU_BIG_ENDIAN depends on
!LD_IS_LLD")

for ARCH=arm in v5.11-rc1.

LLD recently gained support to link ARCH=arm64 BIG_ENDIAN. It does not
yet have support for ARCH=arm, but the Kconfig guards will prevent that
from being selected.  There's some additional complexity to support both
BE32 and BE8 in LLD, but never say never.

Bug: 140224784
Bug: 141733632
Test: BUILD_CONFIG=common/build.config.allmodconfig.aarch64 \
./build/build.sh &&
BUILD_CONFIG=common/build.config.allmodconfig.arm ./build/build.sh
Signed-off-by: Nick Desaulniers <ndesaulniers@google.com>
Change-Id: If24d5ce2a2bbf437c3035e2616db602b61273d02
---
 build.config.allmodconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/build.config.allmodconfig b/build.config.allmodconfig
--- a/build.config.allmodconfig
+++ b/build.config.allmodconfig
@@ -6,7 +6,6 @@ POST_DEFCONFIG_CMDS="update_config"
 function update_config() {
     ${KERNEL_DIR}/scripts/config --file ${OUT_DIR}/.config \
          -d TEST_KMOD  \
-         -d CPU_BIG_ENDIAN \
          -e UNWINDER_FRAME_POINTER \
 
     (cd ${OUT_DIR} && \
